### PR TITLE
Convolutional neural network tutorial

### PR DESCRIPTION
**Pull request by kxk302**
This is ready for review. The tutorial workflow needs the following changes to be deployed before it can be run successfully. 

https://github.com/bgruening/galaxytools/pull/1094
https://github.com/goeckslab/Galaxy-ML/pull/51

The tutorial content, however, can be reviewed now.

### CODE REVIEW

File: topics/statistics/tutorials/CNN/data-library.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
```
Comment (kxk302) [inline topics/statistics/tutorials/CNN/data-library.yaml:1]: Please ignore the file contents here. I will create a new data library after workflow can be run.

**Comment by kxk302** [comment]
ToDo list:

1. Include workflow
2. Upload data to Zenodo and add links in tutorial

File: topics/statistics/tutorials/CNN/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+
+---
+
+
+
+
+# Introduction
+{:.no_toc}
+
+<!-- This is a comment. -->
```
Comment (hexylena) [inline topics/statistics/tutorials/CNN/tutorial.md:32]:
```suggestion
```

On the tutorial formatting side I don't think I have any comments currently, this looks really well formatted from the start! Thanks for preparing it so well.

Comment (kxk302) [reply topics/statistics/tutorials/CNN/tutorial.md]: Thx @hexylena! Applied the change.

File: topics/statistics/tutorials/CNN/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+# Introduction
+{:.no_toc}
+
+
+Artificial neural networks are a machine learning discipline that have been successfully applied to problems
+in pattern classification, clustering, regression, association, time series prediction, optimiztion, and control {% cite JainEtAl %}.
+With the increasing popularity of social media in the past decade, image and video processing tasks have become very
+important. The previous neural network architectures (e.g. feedforward neural networks) could not scale up to handle
+image and video processing tasks. This gave way to the development of convolutional neural networks that are specifically
+tailroed to image and video processing tasks. In this tutorial, we explain what convolutional neural networks are, discuss
```
Comment (anuprulez) [inline topics/statistics/tutorials/CNN/tutorial.md:38]: `tailroed` should be `tailored`

**Review by anuprulez — CHANGES_REQUESTED** [review]

File: topics/statistics/tutorials/CNN/tutorial.md (hunk 3 above)
Comment (hexylena) [reply topics/statistics/tutorials/CNN/tutorial.md:38]: @anuprulez I've made your suggestion into a code suggestion which is easier for contributors to apply :)

File: topics/statistics/tutorials/CNN/data-library.yaml (hunk 1 above)
Comment (kxk302) [reply topics/statistics/tutorials/CNN/data-library.yaml:1]: Contents are good now.

**Comment by hexylena** [comment]
errors are related to the other two PRs not being merged yet, not too worried about that.

File: topics/statistics/tutorials/CNN/data-library.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  type: library
+  name: GTN - Material
+  description: Galaxy Training Network Material
+  synopsis: Galaxy Training Network Material. See https://training.galaxyproject.org
+items:
+- name: Statistics and machine learning
+  description: Statistical Analyses for omics data and machine learning using Galaxy
+    tools
+  items:
+  - name: Introduction to recurrent neural networks (RNN)
```
Comment (qiagu) [inline topics/statistics/tutorials/CNN/data-library.yaml:12]: Convolutional neural network?

File: topics/statistics/tutorials/CNN/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+>        - {% icon param-repeat %} *"6: LAYER"*:
+>            - *"Choose the type of layer"*: `Core -- Flatten`
+>        - {% icon param-repeat %} *"7: LAYER"*:
+>            - *"Choose the type of layer"*: `Core -- Dense`
+>                - *"units"*": `10`
+>                - *"Activation function"*: `softmax`
+>    - Click *"Execute"*
+{: .hands_on}
+
+Each image is passed in as a 784 dimensional vector (28 x 28 = 784). The reshape layer reshapes it into (28, 28, 1) dimensions -- 28 rows, 28 columns, and
```
Comment (qiagu) [inline topics/statistics/tutorials/CNN/tutorial.md:298]: How about width, height, or X, Y?

Comment (kxk302) [reply topics/statistics/tutorials/CNN/tutorial.md]: Revised per fedback.

**Comment by kxk302** [comment]
The PRs in Galaxy-ML and galaxytools have been merged, and the changes have been deployed to galaxy.eu (Thank you to @bgruening  and @qiagu for their help!).
 
 I have also added the workflow and uploaded the datasets to Zenodo.  All is set fot this PR to be reviewed (then merged).
 
 Thanks.

File: topics/statistics/tutorials/CNN/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+>    > * Type `Galaxy Introduction` as the name
+>    > * Press <kbd>Enter</kbd>
+>    >
+>    {: .tip}
+>
+{: .hands_on}
+
+### Data upload
+
+> ### {% icon hands_on %} Hands-on: Data upload
```
Comment (hexylena) [inline topics/statistics/tutorials/CNN/tutorial.md:232]:
```suggestion
```

I'd just merge these into one box.

File: topics/statistics/tutorials/CNN/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+>
+> 1. Create a new history for this tutorial
+> 2. Import the files from [Zenodo](https://zenodo.org/record/)
+>
+>    ```
+>    https://zenodo.org/record/<>/files/X_train.tsv
```
Comment (hexylena) [inline topics/statistics/tutorials/CNN/tutorial.md:238]: You can use page.zenodo_link here to make sure they always stay consistent:

```suggestion
>    {{ page.zenodo_link }}/files/X_train.tsv
```

File: topics/statistics/tutorials/CNN/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+>    {: .tip}
+>
+{: .hands_on}
+
+### Data upload
+
+> ### {% icon hands_on %} Hands-on: Data upload
+>
+> 1. Create a new history for this tutorial
+> 2. Import the files from [Zenodo](https://zenodo.org/record/)
```
Comment (hexylena) [inline topics/statistics/tutorials/CNN/tutorial.md:235]:
```suggestion
> 2. Import the files from [Zenodo]({{ page.zenodo_link }})
```

**Review by hexylena — COMMENTED** [review]
couple small comments on formatting but otherwise this looks good from the formatting side!

File: topics/statistics/tutorials/CNN/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,418 @@
...
+>
+>
+> 1. Create a new history for this tutorial
+> 2. Import the files from [Zenodo]({{ page.zenodo_link }})
+>
+>    ```
+>    {{ page.zenodo_link }}/files/X_train.tsv
+>    https://zenodo.org/record/<>/files/y_train.tsv
+>    https://zenodo.org/record/<>/files/X_test.tsv
+>    https://zenodo.org/record/<>/files/y_test.tsv
```
Comment (hexylena) [inline topics/statistics/tutorials/CNN/tutorial.md:236]: Can you please update these others too?

Comment (kxk302) [reply topics/statistics/tutorials/CNN/tutorial.md]: Done.

**Review by hexylena — APPROVED** [review]
looks good from the formatting side!

**Review by anuprulez — APPROVED** [review]

**Pull request merged**